dev/com.ibm.ws.classloading/src/com/ibm/wsspi/classloading/GatewayConfiguration.java

```diff
@@ -89,6 +89,22 @@
      */
     GatewayConfiguration setApiTypeVisibility(Iterable<ApiType> types);
 
+    /**
+     * Sets the allowed SPI types. 
+     *
+     * @param types The type that of SPIs that are allowed to be loaded by the gateway class loader
+     * @return The GatewayConfiguration with this property set
+     */
+    GatewayConfiguration setSpiTypeVisibility(SpiType... types);
+    
+    /**
+     * Sets the allowed SPI types. 
+     *
+     * @param types The type that of SPIs that are allowed to be loaded by the gateway class loader
+     * @return The GatewayConfiguration with this property set
+     */
+    GatewayConfiguration setSpiTypeVisibility(Iterable<SpiType> types);
```

Comment:
If this is real SPI then we have to throw an exception when it is called for the first time with non-beta?

Reply:
Yes.  Let's add the beta fence, but no-op the beta spi operations.

Reply:
Done.